pkg/operator/operator.go

```diff
@@ -100,6 +108,7 @@ func NewOperator() (context.Context, *Operator) {
 
 	// Inject settings from the ConfigMap(s) into the context
 	ctx = injection.WithSettingsOrDie(ctx, kubernetesInterface, apis.Settings...)
+	opts.MergeSettings(settings.FromContext(ctx))
```

Comment:
The fact that you aren't re-injecting this new options back into the context is _technically_ correct since options is a pointer, but I find it confusing. I'd rather see us just make the call to inject the settings after this function call so that we don't have weird indirection here